ObjRefMsg   ref
string[]    field_names
ObjRefMsg[] field_values